src/ucp/wireup/select.c

```diff
@@ -945,9 +955,9 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_add_memaccess_lanes(
      * remote memory. */
     snprintf(title, sizeof(title), criteria->title, "allocated");
     mem_criteria.title           = title;
-    mem_criteria.local_md_flags  = UCT_MD_FLAG_ALLOC | criteria->local_md_flags;
+    mem_criteria.local_md_flags  = UCT_MD_FLAG_ALLOC | UCT_MD_FLAG_NEED_RKEY |
+                                   criteria->local_md_flags;
```

Comment:
why don't we manage this with tl_bitmap here as well?
looks a bit ungly that now we use both tl_bitmap and mem_criteria to specify mem registration requirements for the lane selections.

Reply:
so you suggest to remove local_md_flags altogether?

Reply:
looks like we can do it

Reply:
it would require more refactoring IMO which would be out of scope of this PR (removing dummy mkey) - today it's also used to require ALLOC and INVALIDATE flags, not just registration.

Reply:
ok